wizard_agent/brain/smart_collection_strategy.py

```diff
@@ -0,0 +1,119 @@
+from typing import List
+import operator
+from . import strategy
+from .utils import util_functions as utils, constants
+
+'''
+Strategy: 
+
+Retreive location of ammo and treasure blocks, 
+find the tile with highest score, and get 
+corresponding ideal path.
+
+'''
+
+
+class SmartCollectionStrategy(strategy.Strategy):
+    def __init__(self):
+    	self.ammo_state = {}
+    	self.prev_bombs = []
+
+        self.game_state = None
+        self.player_state = None
+
+        # value between 0-1, closer to 1 means it's more likely to go to tile
+        self.ammo_priority = 0.5
+        self.treasure_priority = 1
+        self.reachable_priority = 1
+        self.urgent_priority = 1
+
+    def execute(self, game_state: object, player_state: object) -> List[str]:
+        self.game_state = game_state
+        self.player_state = player_state
+        
+        location = player_state.location
+        ammo_blocks = game_state.ammo
+        treasure_blocks = game_state.treasure
+        bombs = game_state.bombs
+
+        exploded_bombs = self.get_exploded_bombs(bombs)
```

Comment:
not sure where this is used but would be good to check if ammo / treasure block is affected by bomb blast zone